cdap-common/src/main/java/co/cask/cdap/common/service/ServiceDiscoverable.java

```diff
@@ -34,6 +36,15 @@ public static String getName(String namespaceId, String applicationId, String se
                          applicationId, serviceId);
   }
 
+  public static ProgramId getId(String name) {
+    String[] parts = name.split("\\.");
```

Comment:
toString, fromString used in EntityId is different from this discoverable naming convention.